File: test_zerocode_app.py
"""Plain app with NO trickle imports.
Instrumented externally: python -m trickle.auto_run test_zerocode_app.py
"""

from test_zerocode_lib import parse_csv, slugify, merge_config

csv_data = parse_csv("name,age,city\nAlice,30,NYC\nBob,25,LA", ",")
print(f"Parsed {csv_data['row_count']} rows")

slug = slugify("Hello World! This is a Test")
print(f"Slug: {slug['slug']}")

config = merge_config(
    {"host": "localhost", "port": 3000, "debug": False},
    {"port": 8080, "debug": True},
)
print(f"Config keys: {config['total_keys']}")

print("Done!")

File: test-zerocode-app.js
/**
 * Plain app with NO trickle code at all.
 * Instrumented externally: node -r trickle/auto test-zerocode-app.js
 */

const { parseCSV, slugify, mergeConfig } = require('./test-zerocode-lib');

const csv = parseCSV("name,age,city\nAlice,30,NYC\nBob,25,LA", ',');
console.log("Parsed", csv.rowCount, "rows");

const slug = slugify("Hello World! This is a Test");
console.log("Slug:", slug.slug);

const config = mergeConfig(
  { host: 'localhost', port: 3000, debug: false },
  { port: 8080, debug: true }
);
console.log("Config keys:", config.totalKeys);

console.log("Done!");
